src/Common/src/CoreLib/System/Numerics/Vector.cs

```diff
@@ -4134,12 +4146,12 @@ internal static unsafe Vector<T> Max(Vector<T> left, Vector<T> right)
             }
         }
 
-        [JitIntrinsic]
+        [Intrinsic]
         internal static T DotProduct(Vector<T> left, Vector<T> right)
         {
             if (Vector.IsHardwareAccelerated)
             {
-                T product = GetZeroValue();
+                T product = default;
```

Comment:
GetZeroValue was throwing an exception for unsupported types. Did that provide any meaningful benefit here, i.e. without it are we losing some protection for something?

Reply:
The exception for unsupported types is still going to be thrown here by `ScalarAdd` a few lines below.

There should be no changes in exceptions thrown from this delta (here and elsewhere). The exception may just be thrown at slightly different place.

Reply:
Cool